src/utils/contract.ts

```diff
@@ -119,4 +117,25 @@ export default class Contract extends BaseConnection {
     });
     return await response.json();
   }
+
+  private async fetchDataViaProvider(params) {
+    switch (this.web3Provider!.providerType) {
+      case "ethers":
+        return await this.web3Provider!
+          .sendAsync('eth_call', params)
+          .then(resp => resp.json()); 
+      case "web3steam":
```

Comment:
Let's call it just `web3`, the stream provider doesn't add anything to the core interface